syntax = "proto3";

package DifferentialTest;

// nested message for message employee
message Company{
  string name = 1;
  string occupation = 2;
  string address = 3;
}
// nested message for message employee
message EducationInfo{
  string name = 1;
  string degree = 2;
  string major = 3;
  string address = 4;
}
// nested message for message employee
message DependentInfo{
  repeated string name = 1;
  repeated int32 age = 2;
}
// nested message for message employee
message ExamScore{
  string exam1 = 1;
  int32 score1 = 2;
  string exam2 = 3;
  float score2 = 4;
}

message TestEmployee{
  int32 employ_id = 1;
  string fullname = 2;
  int32 age = 3;
  Company employer = 4;
  repeated string areas = 5;
  repeated EducationInfo education = 6;
  DependentInfo dependents = 7;
  double floatpoint = 8;

  repeated ExamScore exam_score = 9;

  map<string, string> info = 10;
}

